README.rst

```diff
@@ -0,0 +1,281 @@
+also create connection objects. Connection takes buffer and network
+provider as template parameters as well (note that they must be the same
+as ones of client):
+
+.. code-block:: c
+
+   Connection<Buf_t, Net_t> conn(client);
+
+Connecting
+~~~~~~~~~~
+
+Now assume Tarantool instance is listening ``3301`` port on localhost.
+To connect to the server we should invoke ``Connector::connect()``
+method of client object and pass three arguments: connection instance,
+address and port.
+``int rc = client.connect(conn, address, port)``.
+
+Error handling
+~~~~~~~~~~~~~~
+
+Implementation of connector is exception free, so we rely on return
+codes: in case of fail, ``connect()`` will return ``rc < 0``. To get
+error message corresponding to the last error happened during
+communication with server, we can invoke ``Connection::getError()``
+method:
+
+.. code-block:: c
+
+   if (rc != 0) {
+       assert(conn.status.is_failed);
+       std::cerr << conn.getError() << std::endl;
+   }
+
+To reset connection after errors (clean up error message and connection
+status), one can use ``Connection::reset()``.
+
+Preparing requests
+~~~~~~~~~~~~~~~~~~
+
+To execute simplest request (i.e. ping), one can invoke corresponding
+method of connection object:
+``rid_t ping = conn.ping();``
+
+Each request method returns request id, which is sort of future. It
+can be used to get the result of request execution once it is ready
+(i.e. response). Requests are queued in the input buffer of connection
+until ``Connector::wait()`` is called.
+
+Sending requests
+~~~~~~~~~~~~~~~~
+
+That said, to send requests to the server side, we should invoke
+``client.wait()``:
+
+.. code-block:: c
+
+   client.wait(conn, ping, WAIT_TIMEOUT);
+
+Basically, ``wait()`` takes connection to poll (both IN and OUT),
+request id and optionally timeout (in milliseconds) parameters. once
+response for specified request is ready, ``wait()`` terminates. It also
+provides negative return code in case of system related fails (e.g.
+broken or time outed connection). If ``wait()`` returns 0, then response
+is received and expected to be parsed.
+
+Receiving responses
+~~~~~~~~~~~~~~~~~~~
+
+To get the response when it is ready, we can use
+``Connection::getResponse()``. It takes request id and returns optional
+object containing response (``nullptr`` in case response is not ready
+yet). Note that on each future it can be called only once:
+``getResponse()`` erases request id from internal map once it is
+returned to user.
+
+.. code-block:: c
+
+   std::optional<Response<Buf_t>> response = conn.getResponse(ping);
+
+Response consists of header and body (``response.header`` and
+``response.body``). Depending on success of request execution on server
+side, body may contain either runtime error(s) (accessible by
+``response.body.error_stack``) or data (tuples)
+(``response.body.data``). In turn, data is a vector of tuples. However,
```

Comment:
This paragraph is heavy on parentheses, which make it harder to read. 

Also, it would be good to know the data type of ``error_stack``. Is it also a vector?

```suggestion
Response consists of ``response.header`` and ``response.body``.
In an unsuccessful request, body contains a _____ of runtime errors in 
``response.body.error_stack``.
In a successful request, body contains the request data:
``response.body.data``. Data is a vector of tuples. However,
```